pkg/block/metrics.go

```diff
@@ -0,0 +1,133 @@
+package block
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/httputil"
+)
+
+type MetricsAdapter struct {
```

Comment:
This code could _almost_ be generated by tools/wrapgen from the interface.  I guess I should rip that code out into a separate project, drop the template into a file, and implement directly from there.

Reply:
That's what the template will do, probably by replacing [these lines](https://github.com/treeverse/lakeFS/blob/4dc344001f6df94c7d74556feba00a2ebdf74a37/tools/wrapgen/main.go#L56-L67).